### PR TITLE
[CIT-522] Smagent should use proper mqtt port

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

This PR captures the code changes required for smagent to use the proper mqtt port.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by albinsuresh — COMMENTED** [review]
This PR has a fundamentally different approach to how the port is read from the `TEdgeConfig` compared to #386. These changes make the agent more and more dependent on the `TEdgeConfig` whereas PR #386 is trying to do the exact opposite. So, it would be better if this PR is adapted once #386 is merged.

**Review by didier-wenzek — COMMENTED** [review]
This PR takes a different approach to read the tedge config compared to what is proposed by https://github.com/thin-edge/thin-edge.io/pull/386.

The two PRs must be aligned. I think the approach taken here is less invasive and I advocate for this PR to be taken as the base.

File: sm/tedge_agent/src/agent.rs [hunk 1]
```diff
@@ -97,7 +97,19 @@ impl SmAgent {
...
+            tedge_config::TEdgeConfigRepository::new(self.config_location.clone());
+        let tedge_config = config_repository.load()?;
+
+        let mqtt = Client::connect(
+            self.name.as_str(),
+            &self
+                .config
+                .mqtt_client_config
+                .clone()
+                .with_port(tedge_config.query(MqttPortSetting)?.into()),
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply sm/tedge_agent/src/agent.rs:110]: After discussion with @albinsuresh  and @PradeepKiruvale, the proper place to read the tedge config is in `SmAgentConfig::new(tedge_config_location)` and then to build an agent with `SmAgent::new(sm_agent_config)`.

File: sm/tedge_agent/src/main.rs [hunk 2]
```diff
@@ -19,13 +20,18 @@ async fn main() -> Result<(), anyhow::Error> {
...
     let tedge_config_path = config_repository
         .get_config_location()
         .tedge_config_root_path()
         .to_path_buf();
     let sm_agent_config = SmAgentConfig::default()
         .with_default_plugin_type(default_plugin_type)
-        .with_sm_home(tedge_config_path);
+        .with_sm_home(tedge_config_path)
+        .with_mqtt_client_config(mqtt_config);
+
```
Comment (didier-wenzek) [inline sm/tedge_agent/src/main.rs:34]: Can you please move all this code in a `SmAgentConfig::new(tedge_config_location)` method taking a tedge config as an argument.

**Review by didier-wenzek — CHANGES_REQUESTED** [review]

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**